internal/deploymentcontext/disk_provider.go

```diff
@@ -0,0 +1,195 @@
+		if err != nil {
+			logger.Errorf(err, "Failed to read secret file %s", filePath)
+			continue
+		}
+		// Marshal the file content as a JSON string before storing
+		// This ensures that if the file contains "foo", it's stored as "\"foo\""
+		// so that GetSecret(..., &myString) correctly unmarshals it.
+		jsonEncodedValue, err := json.Marshal(string(fileContentBytes))
+		if err != nil {
+			logger.Errorf(err, "Failed to JSON marshal secret value from file %s", filePath)
+			continue
+		}
+		secrets[entry.Name()] = jsonEncodedValue
+	}
+	return secrets
+}
+
+func NewDiskConfigProvider(path string) ConfigProvider {
+	provider := &diskConfigProvider{
+		configsPath: path,
+	}
+	return provider.GetConfig
+}
+
+// GetConfig reads configs from individual files in the configured directory
+func (p *diskConfigProvider) GetConfig(ctx context.Context) map[string][]byte {
+	logger := log.FromContext(ctx)
+	configs := make(map[string][]byte)
+
+	entries, err := os.ReadDir(p.configsPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			logger.Warnf("Configs directory %s does not exist", p.configsPath)
+			return configs
+		}
+		logger.Errorf(err, "Failed to read configs directory %s", p.configsPath)
+		return configs
+	}
+
+	for _, entry := range entries {
+		if entry.IsDir() || strings.HasPrefix(entry.Name(), ".") { // Skip directories and hidden files
+			continue
+		}
+		filePath := filepath.Join(p.configsPath, entry.Name())
+		fileContentBytes, err := os.ReadFile(filePath)
+		if err != nil {
+			logger.Errorf(err, "Failed to read config file %s", filePath)
+			continue
+		}
+		// Marshal the file content as a JSON string
+		jsonEncodedValue, err := json.Marshal(string(fileContentBytes))
+		if err != nil {
+			logger.Errorf(err, "Failed to JSON marshal config value from file %s", filePath)
+			continue
+		}
+		configs[entry.Name()] = jsonEncodedValue
+	}
+	return configs
+}
+
+// NewDiskRouteProvider creates a new RouteProvider that reads from a schema file on disk
+func NewDiskRouteProvider(ctx context.Context, schemaPath string) (RouteProvider, error) {
+	logger := log.FromContext(ctx)
+
+	provider := &diskRouteProvider{
+		schemaPath: schemaPath,
+	}
+
+	if schemaPath != "" {
+		sch, err := provider.loadSchema() // loadSchema reads from config.SchemaPath
+		if err != nil {
+			logger.Warnf("Failed to load initial schema from %s: %v", schemaPath, err)
+		} else {
+			provider.schema.Store(sch)
+			logger.Debugf("Successfully loaded initial schema from %s", schemaPath)
+		}
+	}
+
+	return provider, nil
+}
+
+func (p *diskRouteProvider) Route(module string) string {
+	sch := p.schema.Load()
+	if sch == nil {
+		return ""
+	}
+
+	routes := make(map[string]string)
+	for _, realm := range sch.Realms {
+		for _, mod := range realm.Modules {
+			for _, verb := range mod.Verbs() {
+				for _, metadata := range verb.Metadata {
+					if ingress, ok := metadata.(*schema.MetadataIngress); ok {
+						routes[mod.Name] = ingress.Method + " " + ingress.PathString()
```

Comment:
Routes are per-module routes, not ingress routes. The template based routing I added is what we actually need here, as it allows is to generate a service name based on the realm / module name.